monai/metrics/average_meter.py

```diff
@@ -0,0 +1,161 @@
+# limitations under the License.
+
+import torch
+import torch.distributed as dist
+
+__all__ = ["AverageMeter"]
+
+
+class AverageMeter:
+    """
+    A utility class to keep track of average values. For example during training/validation loop,
+    we need to accumulate the per-batch metrics and calculate the final average value for the whole dataset.
+    When training in multi-gpu environment, with DistributedDataParallel, it will average across the processes.
+
+    Example:
+
+    .. code-block:: python
+
+        from monai.metrics import AverageMeter
+
+        run_avg = AverageMeter()
+        for i in range(len(train_set)):
+            ...
+            val = calc_metric(x,y) #some metric value
+            run_avg.update(val)
+
+        val_avg = run_avg.get_avg() #average value
+
+    """
+
+    def __init__(self, ensure_cpu_group: bool = False) -> None:
+        """
+        Args:
+            ensure_cpu_group: applies only to multi-gpu distributed setup. Optionally request to operate on CPU,
+                even if the default DistributedDataParallel group is on GPU. This will create an additional
+                CPU group with GLOO backend to reduce across processes on CPU (and maintain all intermediate data on CPU).
+                Defaults to False: use the default process group created during DistributedDataParallel initialization.
+        """
+
+        self.is_distributed = dist.is_available() and dist.is_initialized()
+        self.group = None
+        self.device = None
+
+        if self.is_distributed:
+            dist_cuda = dist.get_backend() == dist.Backend.NCCL
+            if ensure_cpu_group and dist_cuda:
+                self.group = dist.new_group(backend="gloo")  # create a new cpu group
+            elif dist_cuda:
+                self.device = torch.device("cuda", dist.get_rank())
+
+        self.val = 0
+        self.sum = 0
+        self.count = 0
+
+    def proper_tensor(self, x) -> torch.Tensor:
+        """
+        Ensure torch.Tensor float format and optionally copy to the proper gpu device
+        """
+        x = x.detach() if isinstance(x, torch.Tensor) else torch.tensor(x)
+
+        if self.device is not None:
+            x = x.to(device=self.device)
+        else:
+            x = x.cpu()
+
+        return x.float()
+
+    def reduce(self, x: torch.Tensor, avg: bool = False) -> torch.Tensor:
+        """
+        Reduce across processes if in DDP
+        """
+        if self.is_distributed:
+            if avg:
+                x = x / dist.get_world_size()
+            else:
+                x = x.clone()
+            dist.all_reduce(x, group=self.group)
+
+        return x
+
+    def get_current(self, np: bool = True):
+        """
+        return most recent value (averaged across processes)
+        """
+        x = self.reduce(self.val, avg=True)
+        if np:
+            x = x.cpu().numpy()
+        return x
+
+    def get_avg(self, np: bool = True):
```

Comment:
Maybe `to_numpy` instead of `np` to be more descriptive in addition to comments.